### PR TITLE
QA-2477 Migrate ci to ga

### PR DESCRIPTION
**Pull request by paulofachini**
## Description

The purpose of this PR is to migrate the CI tool from Circle CI to Github Actions (Lint, Test and Sonar). Remove these steps from Circleci, Makefile and adjust the Required. Sonar badges will also be added in the README.

Observation:
The goal at this point is to convert to CI pipes only.
CD will be added for later due to credentials to make CD.

## Your checklist for this pull request

:rotating_light: Please review this items for a good pull request. :four_leaf_clover:

1. I've read the project's [Contributing Guidelines](../CONTRIBUTING.md)
1. My commits are well written and follow [pagarme/git-style-guide](https://github.com/pagarme/git-style-guide)
1. My changes are well covered by **tests** and **logs**
1. I've updated the project docs (if needed)
1. I fell safe about this implementation
1. I feel comfortable with the code I wrote, and I'm not ashamed to show it to my friends

In a good pull request, everything above is true :relaxed:

### CODE REVIEW

File: .github/workflows/lint_tests_and_sonar.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      run: npm run lint
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    # - name: Install Docker Compose
```
Comment (Aureojj) [inline .github/workflows/lint_tests_and_sonar.yml:40]: @paulofachini dado que não vamos utilizar esse step do `Install Docker Compose` não seria melhor remover essas linhas?

File: .github/workflows/lint_tests_and_sonar.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    - name: Override Coverage Source Path for Sonar
+      run: |
+        sed -i 's/\/superbowleto\/src\//src\//g' coverage/lcov.info
+    
+    - name: Run Sonar
+      uses: sonarsource/sonarcloud-github-action@master
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
Comment (Aureojj) [inline .github/workflows/lint_tests_and_sonar.yml:87]: @paulofachini O github está reclamando da falta de uma linha em branco no arquivo

File: Makefile [hunk 3]
```diff
@@ -57,7 +57,7 @@ superbowleto-worker:
 	@docker-compose up superbowleto-worker
 .PHONY: superbowleto-worker
 
-sonar:
-	sed -i 's/\/superbowleto\/src\//src\//g' coverage/lcov.info
-	@docker run -ti -v $(shell pwd):/usr/src pagarme/sonar-scanner -Dsonar.branch.name=${BRANCH}
-.PHONY: sonar
+# sonar:
```
Comment (Aureojj) [inline Makefile:60]: @paulofachini Seria o caso de remover essas linhas que estão comentadas no makefile?

File: .github/workflows/lint_tests_and_sonar.yml (hunk 1 above)
Comment (paulofachini) [reply .github/workflows/lint_tests_and_sonar.yml:40]: @Aureojj sim verdade, acabei esquecendo de remover.

File: .github/workflows/lint_tests_and_sonar.yml (hunk 2 above)
Comment (paulofachini) [reply .github/workflows/lint_tests_and_sonar.yml:87]: @Aureojj ajustado.

File: Makefile (hunk 3 above)
Comment (paulofachini) [reply Makefile:60]: @Aureojj verdade, acabei esquecendo de remover.

**Review by DanielGoncalvesL — APPROVED** [review]

**Review by vannessasoares — APPROVED** [review]

**Review by leojasmim — APPROVED** [review]

**Review by JeanCigoli — APPROVED** [review]

**Pull request merged**